src/store/store.controller.ts

```diff
@@ -163,4 +169,73 @@ export class StoreController {
       'Store record updated successfully',
     );
   }
+
+  /**
+   * update store record.
+   * @param {Body} updateStoreDto - Request body object.
+   * @param {Response} res - The payload.
+   * @memberof StoreController
+   * @returns {JSON} - A JSON success response.
+   */
+  @UseGuards(AuthGuard)
+  @Get(':storeId/deliveries')
+  @ApiBadRequestResponse({
+    description: 'Invalid query parameter (offset | period)',
+  })
+  @ApiInternalServerErrorResponse({ description: 'Internal server error' })
+  @ApiParam({ name: 'storeId', required: true, description: 'The store id' })
+  @ApiQuery({
+    name: 'offset',
+    required: false,
+    description: 'Pagination offset',
+    type: 'number',
+  })
+  @ApiQuery({
+    name: 'period',
+    required: false,
+    description: 'Delivery period',
+    enum: ['today', 'upcoming', 'past'],
+  })
+  @ApiQuery({
+    name: 'search',
+    required: false,
+    description: 'Search by team name or producer name',
+    type: 'string',
+  })
+  @ApiHeader({
+    name: 'Authorization',
+    description: 'Bearer <access_token>',
+  })
+  async getStoreDeliveries(
+    @Param('storeId') storeId: string,
+    @Res({ passthrough: true }) res: Response,
+    @Request() req,
+    @Query('offset') offset?: number,
+    @Query('period') period?: 'today' | 'upcoming' | 'past',
+    @Query('search') search?: string,
+  ): Promise<IAPIResponse> {
+    const store = await this.storeService.findStore({ id: storeId });
+    const skip = !isNaN(Number(offset)) ? +offset : 0;
+    if (period && !['today', 'upcoming', 'past'].includes(period))
+      throw new HttpException(
```

Comment:
The formatResponse utility method is used for our response to maintain uniformity